File: services/order-service/src/main/java/io/minishop/order/inbox/InventoryInboxRepository.java
package io.minishop.order.inbox;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface InventoryInboxRepository extends JpaRepository<InventoryInboxRecord, Long> {
    boolean existsByReservationId(Long reservationId);
    List<InventoryInboxRecord> findByOrderId(Long orderId);
}

File: services/order-service/src/main/kotlin/io/minishop/order/inbox/InventoryInboxRepository.kt
package io.minishop.order.inbox

import org.springframework.data.jpa.repository.JpaRepository

interface InventoryInboxRepository : JpaRepository<InventoryInboxRecord, Long> {
    fun existsByReservationId(reservationId: Long): Boolean
    fun findByOrderId(orderId: Long): List<InventoryInboxRecord>
}
